microsoft-edge/webview2/concepts/working-with-local-content.md

```diff
@@ -282,14 +396,49 @@ webView->Navigate(L"https://demo/index.html");
 
 `WebResourceRequested` allows you to customize the behavior of local content on a per-request basis. This means you can decide which requests to intercept and provide your own content for, and which requests to let the WebView2 control handle normally.  However, customizing the behavior requires more code, such as virtual host mapping, and requires knowledge of HTTP, to be able to construct a proper response. 
 
 From WebView2's perspective, the resource will have come via the network, and WebView2 will adhere to the headers that are set by the app as part of the response. Using the `WebResourceRequested` event is also slower than other approaches, due to the needed cross-process communication and processing per each request.
 
+
+
+<!-- ------------------------------ -->
+#### Considerations for loading local content by handling the WebResourceRequested event
+
+
+###### Origin-based DOM APIs
+
+Local content loaded via `WebResourceRequested` has an http or https URL and a corresponding origin. So APIs that require an origin such as `localStorage` or `indexedDB` will work, and the usual ["same-origin" rules](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) apply.
+
+
+###### DOM APIs requiring secure context
+
+Some newer browser features are limited to https URLs. Using `WebResourceRequested` allows you to replace https URL web resource requests with your own local content, and so these browser features are available. This means APIs such as webcam APIs, geolocation APIs, and notification APIs are all available. For more information, see [Secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts).
+
+
+###### Dynamic content
+
+When loading local content via `WebResourceRequested`, you specify the local content to load in your event handler. Since you are directly providing the local content, you can dynamically modify it in any manner.
+
+
+###### Additional web resources
+
+Local content that's loaded via `WebResourceRequested` modifies content that's loaded via http and https URLs. These URLs support relative URL resolution, and so an HTML document that's served via `WebResourceRequested` can have references such as CSS, script, and image references that are also served via `WebResourceRequested`.
+
+
+###### Additional web resources resolved in WebView2 process
+
+Unlike file URLs and virtual host name mappings, which are resolved in WebView2 processes, the `WebResourceRequested` event is raised on your WebView2's UI thread in your host app process. This means the WebView2 will first pause loading a web page to wait for the event to be sent to your host app process, and then wait for your UI thread to be available, and then wait for your app code to handle the event. This can take some time, so make sure that your calls to `AddWebResourceRequestedFilter` are appropriately limited to only the web resources that must raise the `WebResourceRequested` event.
```

Comment:
```suggestion
Unlike file URLs and virtual host name mappings, which are resolved in WebView2 processes, the `WebResourceRequested` event is raised on your WebView2's UI thread in your host app process. This means the WebView2 will first pause loading a web page to wait for the event to be sent to your host app process, and then wait for your UI thread to be available, and then wait for your app code to handle the event. This can take some time, so make sure that your calls to `AddWebResourceRequestedFilter` are limited to only the web resources that must raise the `WebResourceRequested` event.
```
judgment call.  i stumble on "appropriately", distracting, seems redundant, raises more complexity, not a net gain by adding that word?